clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobPoller.java

```diff
@@ -133,6 +131,24 @@ public void onCompleted() {
     pollingDone();
   }
 
+  private void logFailure(final Throwable throwable) {
+    final String errorMsg = "Failed to activated jobs for worker {} and job type {}";
+
+    if (throwable instanceof StatusRuntimeException) {
+      final StatusRuntimeException statusRuntimeException = (StatusRuntimeException) throwable;
+      if (statusRuntimeException.getStatus().equals(Status.RESOURCE_EXHAUSTED)) {
```

Comment:
From the docs of `Status#equals`:

> Equality on Statuses is not well defined. Instead, do comparison based on their Code with getCode. The description and cause of the Status are unlikely to be stable, and additional fields may be added to Status in the future.

```suggestion
      if (statusRuntimeException.getStatus().getCode() == Code.RESOURCE_EXHAUSTED) {
```